src/navier_stokes/INSVCStaggeredHierarchyIntegrator.cpp

```diff
@@ -1758,7 +1758,7 @@ INSVCStaggeredHierarchyIntegrator::applyGradientDetectorSpecialized(const Pointe
         d_hier_math_ops->curl(d_Omega_idx, d_Omega_var, d_U_scratch_idx, d_U_var, d_U_bdry_bc_fill_op, error_data_time);
         tagCellsByVorticityMagnitude(level_number, d_Omega_idx, tag_index);
 
-        for (int ln = 0; ln < d_hierarchy->getFinestLevelNumber(); ++ln)
+        for (int ln = 0; ln <= d_hierarchy->getFinestLevelNumber(); ++ln)
```

Comment:
Why do we need to evaluate the gradient detector on the finest level of the patch hierarchy?

Reply:
We could fix this in a follow-up, but `INSVCStaggeredHierarchyIntegrator` assumes that the finest patch level is allocated. I'd prefer to leave that as-is for now - allocating one more vector once isn't that big of a deal.